src/core/stdc/math.d

```diff
@@ -262,18 +284,63 @@ else version( CRuntime_Microsoft )
     int _isnan(double x);
     ///
     int _fpclass(double x);
+  }
+
+  extern(D)
+  {
+    //int fpclassify(real-floating x);
+    ///
+    int fpclassify(float x)     { return _fdclass(x); }
+    ///
+    int fpclassify(double x)    { return _dclass(x);  }
+    ///
+    int fpclassify(real x)      { static if (real.sizeof == double.sizeof) return _dclass(x); else assert(0); }
+
+    //int isfinite(real-floating x);
+    ///
+    int isfinite(float x)       { return fpclassify(x) <= 0; }
+    ///
+    int isfinite(double x)      { return fpclassify(x) <= 0; }
+    ///
+    int isfinite(real x)        { return fpclassify(x) <= 0; }
+
+    //int isinf(real-floating x);
+    ///
+    int isinf(float x)          { return fpclassify(x) == FP_INFINITE; }
+    ///
+    int isinf(double x)         { return fpclassify(x) == FP_INFINITE; }
+    ///
+    int isinf(real x)           { return fpclassify(x) == FP_INFINITE; }
+
+    //int isnan(real-floating x);
+    ///
+    int isnan(float x)          { return fpclassify(x) == FP_NAN; }
+    ///
+    int isnan(double x)         { return fpclassify(x) == FP_NAN; }
+    ///
+    int isnan(real x)           { return fpclassify(x) == FP_NAN; }
```

Comment:
How about implementing isnan by calling _isnan or _fpclass and have backward compatibility for symbols declared in math.d until now?